app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
+      persistent: true,
+      connect_timeout: TIMEOUT)
+  end
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    route = 'spamreports.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
```

Comment:
Generally not a fan of catching all exceptions... I don't think it's necessary in this case.